rai/node/rpc.cpp

```diff
@@ -889,6 +889,127 @@ void rai::rpc_handler::accounts_pending ()
 	response (response_l);
 }
 
+void rai::rpc_handler::sign_message ()
+{
+	rai::uint256_union account (0);
+	boost::optional<std::string> account_text (request.get_optional<std::string> ("account"));
+	if (account_text.is_initialized ())
+	{
+		auto error_account (account.decode_account (account_text.get ()));
+		if (error_account)
+		{
+			error_response (response, "Bad account number");
+		}
+	}
+	rai::uint256_union wallet (0);
+	boost::optional<std::string> wallet_text (request.get_optional<std::string> ("wallet"));
+	if (wallet_text.is_initialized ())
+	{
+		auto error (wallet.decode_hex (wallet_text.get ()));
+		if (error)
+		{
+			error_response (response, "Bad wallet number");
```

Comment:
One thing we're trying to avoid is having `error_response` be called twice. IIRC it causes a double free. That'll happen here if neither account nor wallet is initialized. Please check `!error && wallet_text.is_initialized ()` above (line 906 at time of writing).